pkg/kubectl-nfd/test.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectlnfd
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	taintutils "k8s.io/kubernetes/pkg/util/taints"
+
+	nfdv1alpha1 "sigs.k8s.io/node-feature-discovery/pkg/apis/nfd/v1alpha1"
+	nfdclientset "sigs.k8s.io/node-feature-discovery/pkg/generated/clientset/versioned"
+	"sigs.k8s.io/yaml"
+)
+
+func Test(nodefeaturerulepath, nodeName, kubeconfig, namespace string) []error {
+	var validationErr []error
+	var nodes []corev1.Node
+	var node corev1.Node
+	var err error
+	var clientset *kubernetes.Clientset
+
+	nfr := nfdv1alpha1.NodeFeatureRule{}
+
+	if kubeconfig == "" {
+		kubeconfig = os.Getenv("KUBECONFIG")
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return []error{fmt.Errorf("error building kubeconfig: %v", err)}
+	}
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		return []error{fmt.Errorf("error building kubernetes clientset: %v", err)}
+	}
+	cli := nfdclientset.NewForConfigOrDie(config)
+
+	nodes, err = getNonControlPlaneNodes(clientset)
+	if err != nil {
+		return []error{fmt.Errorf("error getting non-control-plane nodes: %v", err)}
+	}
+	if nodeName != "" {
+		node = getNode(nodes, nodeName)
+	} else {
+		// If no node is specified, use the first non-control-plane node
+		node = nodes[0]
+	}
+
+	nf, err := cli.NfdV1alpha1().NodeFeatures(namespace).Get(context.Background(), node.Name, metav1.GetOptions{})
+	if err != nil {
+		return []error{fmt.Errorf("error getting NodeFeature: %v", err)}
+	}
+
+	nfrFile, err := os.ReadFile(nodefeaturerulepath)
+	if err != nil {
+		return []error{fmt.Errorf("error reading NodeFeatureRule file: %v", err)}
+	}
+
+	err = yaml.Unmarshal(nfrFile, &nfr)
+	if err != nil {
+		return []error{fmt.Errorf("error reading NodeFeatureRule file: %v", err)}
+	}
+
+	validationErr = append(validationErr, processNodeFeatureRule(nfr, *nf)...)
+
+	return validationErr
```

Comment:
Looks like used first time here so just

```suggestion
	return processNodeFeatureRule(nfr, *nf)
```